velox/functions/sparksql/String.h

```diff
@@ -1431,4 +1432,212 @@ struct LevenshteinDistanceFunction {
   }
 };
 
+// mask(string) -> string
+// mask(string, upperChar) -> string
+// mask(string, upperChar, lowerChar) -> string
+// mask(string, upperChar, lowerChar, digitChar) -> string
+// mask(string, upperChar, lowerChar, digitChar, otherChar) -> string
+//
+// Masks the characters of the given string value with the provided specific
+// characters respectively. Upper-case characters are replaced with the second
+// argument. Default value is 'X'. Lower-case characters are replaced with the
+// third argument. Default value is 'x'. Digit characters are replaced with the
+// fourth argument. Default value is 'n'. Other characters are replaced with the
+// last argument. Default value is NULL and the original character is retained.
+// If the provided nth argument is NULL, the related original character is
+// retained.
+template <typename T>
+struct MaskFunction {
```

Comment:
Would it make sense to provide fast path for ASCII inputs?

Reply:
emm.. looks leverage `callASCII` have benefit but not much since we still need handle `replacement char` args is non ASCII.
Create a issue  to do in seperate PR if needed. https://github.com/facebookincubator/velox/issues/10546

Reply:
@mbasmanova It seems only `callAscii` is provided, while a function like `callAsciiNullable` is needed here. Do you think we need to add that? Thanks.